### ISSUE TITLE
[BUG]: Nsight Compute bug with Pytorch +cuTile

### ISSUE THREAD

**Issue by PeppeRu96** [NVIDIA/cutile-python#31]
### Version

1.0.0

### Version

13.1

torch.version
'2.9.1+cu130'

cupy.version
'13.6.0'

### Which installation method(s) does this occur on?

Pip

### Describe the bug.

I'm running on an RTX5090 with Cuda 13.1, Driver: 590.44.01.
I'm basically getting:
```
=PROF== Profiling "distribution_elementwise_grid..." - 0: 0%....50%....100% - 40 passes
==PROF== Profiling "distribution_elementwise_grid..." - 1: 0%....50%....100% - 40 passes
==PROF== Profiling "vec_add_kernel_1d" - 2: 0%....50%....100% - 40 passes
✓ vector_add_example passed!
double free or corruption (!prev)
==ERROR== The application returned an error code (6).
```

The error arises from double cuda free (it seems). This doesn't happen with cupy tensors.

### Minimum reproducible example

```shell
import cupy as cp
import numpy as np
import cuda.tile as ct
import torch
import triton

DEVICE = triton.runtime.driver.active.get_active_torch_device()


@ct.kernel
def vector_add(a, b, c, tile_size: ct.Constant[int]):
    # Get the 1D pid
    pid = ct.bid(0)

    # Load input tiles
    a_tile = ct.load(a, index=(pid,), shape=(tile_size,))
    b_tile = ct.load(b, index=(pid,), shape=(tile_size,))

    # Perform elementwise addition
    result = a_tile + b_tile

    # Store result
    ct.store(c, index=(pid, ), tile=result)

def test_torch():
    vector_size = 2**12
    tile_size = 2**4
    grid = (ct.cdiv(vector_size, tile_size), 1, 1)

    a = torch.empty(vector_size, device=DEVICE, dtype=torch.float16).normal_(mean=0.3, std=0.2)
    b = torch.empty(vector_size, device=DEVICE, dtype=torch.float16).normal_(mean=0.3, std=0.2)
    c = torch.empty_like(a)

    # Launch kernel
    ct.launch(torch.cuda.current_stream(),
              grid,  # 1D grid of processors
              vector_add,
              (a, b, c, tile_size))
    

    torch.cuda.synchronize()

    expected = a + b
    torch.testing.assert_close(c, expected, rtol=1e-6, atol=1e-6)

    print("✓ vector_add_example passed!")

test_torch()


Run: `ncu -o VecAddProfile --set detailed python3 VectorAdd_quickstart.py`
```

### Relevant log output

```shell
=PROF== Profiling "distribution_elementwise_grid..." - 0: 0%....50%....100% - 40 passes
==PROF== Profiling "distribution_elementwise_grid..." - 1: 0%....50%....100% - 40 passes
==PROF== Profiling "vec_add_kernel_1d" - 2: 0%....50%....100% - 40 passes
✓ vector_add_example passed!
double free or corruption (!prev)
==ERROR== The application returned an error code (6).
```

### Full env printout

<details><summary>Click here to see environment details</summary><pre>
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=24.04
     DISTRIB_CODENAME=noble
     DISTRIB_DESCRIPTION="Ubuntu 24.04.3 LTS"
     PRETTY_NAME="Ubuntu 24.04.3 LTS"
     NAME="Ubuntu"
     VERSION_ID="24.04"
     VERSION="24.04.3 LTS (Noble Numbat)"
     VERSION_CODENAME=noble
     ID=ubuntu
     ID_LIKE=debian
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     UBUNTU_CODENAME=noble
     LOGO=ubuntu-logo
     Linux lukas 6.14.0-37-generic #37~24.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 20 10:25:38 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Thu Dec 18 19:21:01 2025
     +-----------------------------------------------------------------------------------------+
     | NVIDIA-SMI 590.44.01              Driver Version: 590.44.01      CUDA Version: 13.1     |
     +-----------------------------------------+------------------------+----------------------+
     | GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
     |                                         |                        |               MIG M. |
     |=========================================+========================+======================|
     |   0  NVIDIA GeForce RTX 5090        On  |   00000000:01:00.0 Off |                  N/A |
     |  0%   37C    P8             18W /  560W |      15MiB /  32607MiB |      0%      Default |
     |                                         |                        |                  N/A |
     +-----------------------------------------+------------------------+----------------------+
     
     +-----------------------------------------------------------------------------------------+
     | Processes:                                                                              |
     |  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
     |        ID   ID                                                               Usage      |
     |=========================================================================================|
     |    0   N/A  N/A            1820      G   /usr/lib/xorg/Xorg                        4MiB |
     +-----------------------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:                            x86_64
     CPU op-mode(s):                          32-bit, 64-bit
     Address sizes:                           48 bits physical, 48 bits virtual
     Byte Order:                              Little Endian
     CPU(s):                                  32
     On-line CPU(s) list:                     0-31
     Vendor ID:                               AuthenticAMD
     Model name:                              AMD Ryzen 9 9950X3D 16-Core Processor
     CPU family:                              26
     Model:                                   68
     Thread(s) per core:                      2
     Core(s) per socket:                      16
     Socket(s):                               1
     Stepping:                                0
     Frequency boost:                         enabled
     CPU(s) scaling MHz:                      64%
     CPU max MHz:                             5756.0000
     CPU min MHz:                             600.0000
     BogoMIPS:                                8599.98
     Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx_vnni avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd cppc arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid bus_lock_detect movdiri movdir64b overflow_recov succor smca fsrm avx512_vp2intersect flush_l1d amd_lbr_pmc_freeze
     Virtualization:                          AMD-V
     L1d cache:                               768 KiB (16 instances)
     L1i cache:                               512 KiB (16 instances)
     L2 cache:                                16 MiB (16 instances)
     L3 cache:                                128 MiB (2 instances)
     NUMA node(s):                            1
     NUMA node0 CPU(s):                       0-31
     Vulnerability Gather data sampling:      Not affected
     Vulnerability Ghostwrite:                Not affected
     Vulnerability Indirect target selection: Not affected
     Vulnerability Itlb multihit:             Not affected
     Vulnerability L1tf:                      Not affected
     Vulnerability Mds:                       Not affected
     Vulnerability Meltdown:                  Not affected
     Vulnerability Mmio stale data:           Not affected
     Vulnerability Reg file data sampling:    Not affected
     Vulnerability Retbleed:                  Not affected
     Vulnerability Spec rstack overflow:      Mitigation; IBPB on VMEXIT only
     Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
     Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
     Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; PBRSB-eIBRS Not affected; BHI Not affected
     Vulnerability Srbds:                     Not affected
     Vulnerability Tsa:                       Not affected
     Vulnerability Tsx async abort:           Not affected
     Vulnerability Vmscape:                   Mitigation; IBPB on VMEXIT
     
     ***CMake***
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
     Copyright (C) 2023 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda-13.1/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2025 NVIDIA Corporation
     Built on Fri_Nov__7_07:23:37_PM_PST_2025
     Cuda compilation tools, release 13.1, V13.1.80
     Build cuda_13.1.r13.1/compiler.36836380_0
     
     ***Python***
     /home/ruggeri/prj/.venv/bin/python
     Python 3.12.3
     
     ***Environment Variables***
     PATH                            : /home/ruggeri/prj/.venv/bin:/usr/local/cuda-13.1/bin:/home/ruggeri/scripts:/home/ruggeri/.vscode-server/data/User/globalStorage/github.copilot-chat/debugCommand:/home/ruggeri/.vscode-server/data/User/globalStorage/github.copilot-chat/copilotCli:/home/ruggeri/.vscode-server/cli/servers/Stable-1e3c50d64110be466c0b4a45222e81d2c9352888/server/bin/remote-cli:/home/ruggeri/.local/bin:/usr/local/cuda-13.1/bin:/home/ruggeri/.local/bin:/home/ruggeri/scripts:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ruggeri/.vscode-server/extensions/ms-python.debugpy-2025.18.0-linux-x64/bundled/scripts/noConfigScripts
     LD_LIBRARY_PATH                 : /usr/local/cuda-13.1/lib64:/usr/local/cuda-13.1/lib64:
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    :
     PYTHON_PATH                     :
     
     conda not found
     ***pip packages***
     /usr/bin/pip
     Package               Version
     --------------------- ---------------
     attrs                 23.2.0
     Babel                 2.10.3
     bcc                   0.29.1
     blinker               1.7.0
     Brlapi                0.8.5
     certifi               2023.11.17
     chardet               5.2.0
     click                 8.1.6
     cloud-init            25.2
     colorama              0.4.6
     command-not-found     0.3
     configobj             5.0.8
     cryptography          41.0.7
     cupshelpers           1.0
     dbus-python           1.3.2
     defer                 1.0.6
     distro                1.9.0
     distro-info           1.7+build1
     httplib2              0.20.4
     idna                  3.6
     Jinja2                3.1.2
     jsonpatch             1.32
     jsonpointer           2.0
     jsonschema            4.10.3
     language-selector     0.1
     launchpadlib          1.11.0
     lazr.restfulclient    0.14.6
     lazr.uri              1.0.6
     louis                 3.29.0
     markdown-it-py        3.0.0
     MarkupSafe            2.1.5
     mdurl                 0.1.2
     netaddr               0.8.0
     netifaces             0.11.0
     oauthlib              3.2.2
     olefile               0.46
     pexpect               4.9.0
     pillow                10.2.0
     pip                   24.0
     ptyprocess            0.7.0
     pycairo               1.25.1
     pycups                2.0.1
     Pygments              2.17.2
     PyGObject             3.48.2
     PyJWT                 2.7.0
     pyparsing             3.1.1
     pyrsistent            0.20.0
     pyserial              3.5
     python-apt            2.7.7+ubuntu5.1
     python-dateutil       2.8.2
     python-debian         0.1.49+ubuntu2
     pytz                  2024.1
     pyxdg                 0.28
     PyYAML                6.0.1
     requests              2.31.0
     rich                  13.7.1
     setuptools            68.1.2
     six                   1.16.0
     ssh-import-id         5.11
     systemd-python        235
     typing_extensions     4.10.0
     ubuntu-drivers-common 0.0.0
     ubuntu-pro-client     8001
     ufw                   0.36.2
     unattended-upgrades   0.1
     urllib3               2.0.7
     wadllib               1.3.6
     wheel                 0.42.0
     xdg                   5
     xkit                  0.0.0
     
</pre></details>

### Contributing Guidelines

- [x] I agree to follow cuTile Python's contributing guidelines
- [x] I have searched the [open bugs](https://github.com/nvidia/cutile-python/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by haijieg** [comment]
> > > > [@PeppeRu96](https://github.com/PeppeRu96) can you share out put of "bash print_env.sh"?
> > > 
> > > 
> > > Done in the original issue message.
> > 
> > 
> > The environment doesn't show torch version or cupy. How's your pytorch / cupy installed and what version are they?
> 
> I installed cupy from pip.
> 
> torch.**version** '2.9.1+cu130'
> 
> cupy.**version** '13.6.0'

Strange. I wonder why that didn't get printed by "print_env.sh" which use "pip list" for pip environment.
